extensions/adhan-time/utils/sortPrayers.ts

```diff
@@ -0,0 +1,20 @@
+import { Prayer } from "../src/types/prayerTypes";
+
+export function sortTimingsByTypeAndTime(prayers: Prayer[]): Prayer[] {
+  return [...prayers].sort((a, b) => {
+    // First, separate prayers from timings
+    const aIsPrayer = a.type === "prayer";
+    const bIsPrayer = b.type === "prayer";
```

Comment:
do you mean timing?

Reply:
The role of this method is to sort prayers first then timings. `aIsPrayer` and `bIsPrayer` are used as predicates so that if `a` is a prayer and `b` is a timing it returns `a` first, if both are the same whether prayers or timings, then they get sorted chronologically.